fdt.hh

```diff
@@ -713,6 +728,12 @@ class device_tree
 	 * These should be expanded to the full path of their targets.
 	 */
 	std::vector<property_value*> cross_references;
+	/**
+	 * A collection of nodes that are referenced in this device tree.
+	 * These are later checked when we walk the tree to remove any nodes that
+	 * are marked for deletion if they're later found to be unreferenced.
+	 */
+	std::vector<node*> referenced_nodes;
```

Comment:
This probably wants to be a `std::unordered_set` - you don't care if a thing is referenced more than once.

Reply:
It probably also doesn't need to be a field - you should do the GC scan and deletion in a single function and just make this a local temporary.